assets/src/edit-story/components/mediaPicker/test/useMediaPicker.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { renderHook } from '@testing-library/react-hooks';
+
+/**
+ * Internal dependencies
+ */
+import ConfigContext from '../../../app/config/context';
+import useMediaPicker from '../useMediaPicker';
+import SnackbarContext from '../../../app/snackbar/context';
+
+jest.mock('../../../app/snackbar', () => ({
+  useSnackbar: () => {
+    return {
+      showSnackbar: jest.fn(),
+    };
+  },
+}));
+
+function setup(args) {
+  const configValue = {
+    capabilities: {
+      hasUploadMediaAction: true,
+    },
+    ...args,
+  };
+  const wrapper = (params) => (
+    <ConfigContext.Provider value={configValue}>
+      <SnackbarContext.Provider>{params.children}</SnackbarContext.Provider>
+    </ConfigContext.Provider>
+  );
+  const onSelect = jest.fn();
+  const onClose = jest.fn();
+  const { result } = renderHook(() => useMediaPicker({ onSelect, onClose }), {
+    wrapper,
+  });
+  return {
+    openMediaPicker: result.current,
+  };
+}
+
+describe('useMediaPicker', () => {
+  it('user unable to upload', () => {
+    const { openMediaPicker } = setup({
+      capabilities: {
+        hasUploadMediaAction: false,
+      },
+    });
+    const evt = {
+      preventDefault: jest.fn(),
+    };
+    expect(openMediaPicker(evt)).toBe(false);
```

Comment:
`openMediaPicker` always returns `false` though.

Might make more sense to test that `wp.media` hasn't been called.

Reply:
That also means you'd want a successful test case where `wp.media` _is being called_.

Reply:
Do you have any advice on mocking out wp.media. I just tried and it ending up being a massive rubbit hole.